### PR TITLE
Added price filter

### PR DESCRIPTION
**Pull request by brettallenyo**
Added a filter that allows the user to filter through the stores based on price using both a slider as well as a text input

The slider and text input adapt the the maximum cost of the original stores, and the standard step allotted for the text input (when you hover over the text input there is an arrow up and arrow down button) is the maximum price divided by 10.

EDIT: Made it so that if you only pass one item type the filter for items will not show up at all

Screen Shot 2020-07-22 at 1 49 45 PM

--------------------------------------------------------------------------------------------------------------------

Screen Shot 2020-07-22 at 1 49 51 PM

--------------------------------------------------------------------------------------------------------------------

Screen Shot 2020-07-22 at 1 49 59 PM

+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

Added feature so that you can set minimum price, also fixed bug for getting the user lists

Screen Shot 2020-07-22 at 9 10 50 PM

________________________________________________________________________________________________________________________
________________________________________________________________________________________________________________________
________________________________________________________________________________________________________________________

Also fixed bug to get user lists when user logs in after the page has already been fully rendered

Screen Shot 2020-07-23 at 11 01 09 AM

### CODE REVIEW

File: capstone/client/src/components/FilterStores.js [hunk 1]
```diff
@@ -82,10 +139,50 @@ class FilterStores extends Component {
...
+          <Grid item xs>
+            <Slider
+                  value={this.state.setPrice}
+                  onChange={this.onPriceChange}
+                  aria-labelledby="input-slider"
+                  color="action"
+                  max={this.state.maxPrice}
+            />
+          </Grid>
+          <Grid item>
```
Comment (chen-dawn) [inline capstone/client/src/components/FilterStores.js:168]: do we need so many grid?

Comment (brettallenyo) [reply capstone/client/src/components/FilterStores.js:168]: They had all these grids in the documentation so I just kinda trust the process, I can try to change it though if you want?

File: capstone/client/src/components/FilterStores.js [hunk 2]
```diff
@@ -82,10 +154,67 @@ class FilterStores extends Component {
...
+              value={this.state.setPriceLeft}
+              margin="dense"
+              onChange={this.handleInputChangeLeft}
+              onBlur={this.handleBlur}
+              inputProps={{
+                step: this.state.maxPrice/40,
+                min: 0,
+                max: this.state.maxPrice,
+                type: 'number',
+                'aria-labelledby': 'input-slider',
```
Comment (chen-dawn) [inline capstone/client/src/components/FilterStores.js:184]: do we need the quotes here?

Comment (brettallenyo) [reply capstone/client/src/components/FilterStores.js:184]: yeah its a really weird prop

**Pull request merged**